### PR TITLE
chore(ci): bump package version after release via pull request

### PR DESCRIPTION
**Pull request by heitorlessa**
**Issue number:** #2210

**DEPENDENCY**: https://github.com/awslabs/aws-lambda-powertools-python/pull/2238

## Summary

Use new create-pr action introduced in https://github.com/awslabs/aws-lambda-powertools-python/pull/2238 to bump package version via PR instead of pushing directly to `develop` as part of the release.

### Changes

> Please provide a summary of what's being changed

* [x] Create new job `bump_version` with scoped permissions to do exactly that
* [x] Lower permission in `build` job as it no longer needs to push to `develop`
* [x] Remove `changelog` job as it's now handled perfectly in every push via PR - introduced in https://github.com/awslabs/aws-lambda-powertools-python/pull/2238

### User experience

> Please share what the user experience looks like before and after this change

## Checklist

If your change doesn't seem to apply, please leave them unchecked.

* [x] [Meet tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
* [x] I have performed a self-review of this change
* [x] Changes have been tested
* [x] Changes are documented
* [x] PR title follows [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-python/blob/develop/.github/semantic.yml)

<details>
<summary>Is this a breaking change?</summary>

**RFC issue number**:

Checklist:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

</details>

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -163,6 +140,28 @@ jobs:
       latest_published_version: ${{ needs.build.outputs.RELEASE_VERSION }}
       pre_release: ${{ inputs.pre_release }}
 
+  bump_version:
+    needs: [build, release]
+    permissions:
+      contents: write
+      pull-requests: write
```
Comment (leandrodamascena) [inline .github/workflows/release.yml]: Are these permissions not conflicting with each other? I think the `pull-requests: write` permission is enough for that, no? The `contents: write` permission seems to be much broader and allows things we don't need.

References:
https://docs.github.com/en/rest/overview/permissions-required-for-github-apps?apiVersion=2022-11-28#pull-requests
https://docs.github.com/en/rest/overview/permissions-required-for-github-apps?apiVersion=2022-11-28#contents

**Review by leandrodamascena — CHANGES_REQUESTED** [review]
I just left a comment about permissions, but for everything else, this looks good!

File: .github/workflows/release.yml (hunk 1 above)
Comment (heitorlessa) [reply .github/workflows/release.yml]: Not in this case :) 

**`create-pr`** action needs to create a temporary branch to push these staged files (`pyproject.toml`) - hence `contents: write`. Then, creates a PR against `develop` using our temporary branch as the HEAD - hence `pull-requests: write`

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -163,6 +140,28 @@ jobs:
       latest_published_version: ${{ needs.build.outputs.RELEASE_VERSION }}
       pre_release: ${{ inputs.pre_release }}
 
+  bump_version:
+    needs: [build, release]
+    permissions:
+      contents: write
```
Comment (heitorlessa) [inline .github/workflows/release.yml]:
```suggestion
      contents: write  # create-pr action creates a temporary branch
```

Comment (heitorlessa) [inline .github/workflows/release.yml]:
```suggestion
      contents: write  # create-pr action creates a temporary branch
```

File: .github/workflows/release.yml (hunk 1 above)
Comment (heitorlessa) [inline .github/workflows/release.yml]:
```suggestion
      pull-requests: write # create-pr action creates a PR using the temporary branch
```

**Review by leandrodamascena — APPROVED** [review]
LGTM!

**Pull request merged**